lib/datasets/rdatasets.rb

```diff
@@ -82,23 +82,52 @@ def each(&block)
       return to_enum(__method__) unless block_given?
 
       download(@data_path, @metadata.url)
+
       symbol_raw_converter = lambda do |header|
-        header.encode(CSV::ConverterEncoding).to_sym
+        # temporally code for "validate/nace_rev2"
+        if [@package_name, @dataset_name] == %w[validate nace_rev2]
+          header.encode(CSV::ConverterEncoding).to_sym
+        else
+          header.encode(CSV::ConverterEncoding)[1..-2].to_sym # always over-quoted by quote_char: nil
+        end
       end
+
       na_converter = lambda do |field|
-        begin
-          if field.encode(CSV::ConverterEncoding) == "NA"
-            nil
-          else
-            field
-          end
-        rescue
-          field
+        field.encode(CSV::ConverterEncoding) == "NA" ? nil : field rescue field
+      end
+
+      inf_converter = lambda do |field|
+        field.encode(CSV::ConverterEncoding) == "Inf" ? Float::INFINITY : field rescue field
+      end
+
+      # This must be the last converter
+      quote_preserving_converter = lambda do |field, field_info|
+        f = field.encode(CSV::ConverterEncoding)
+
+        # temporally code for "validate/nace_rev2"
+        if [@package_name, @dataset_name] == %w[validate nace_rev2]
+          quoted = {Level: false, Order: false}
+          return f if quoted.fetch(field_info.header, true)
         end
+
+        m = /"(.+)"/.match(f)
```

Comment:
```suggestion
        m = /\A"(.+)"\z/.match(f)
```

The following code may be faster than it:

```ruby
  return f[1..-2] if f.start_with?('"') and f.end_with?('"')
```